trieste/acquisition/optimizer.py

```diff
@@ -65,16 +75,25 @@ def _discrete_space(space: DiscreteSearchSpace, target_func: AcquisitionFunction
 
 
 @optimize.register
-def _box(space: Box, target_func: AcquisitionFunction) -> TensorType:
-    trial_search_space = space.discretize(tf.minimum(2000, 500 * tf.shape(space.lower)[-1]))
-    initial_point = optimize(trial_search_space, target_func)
+def optimize_continuous(
+    space: Box, target_func: AcquisitionFunction, num_query_points: int = 1
+) -> TensorType:
+    expanded_space = space ** num_query_points
+    trial_search_space = expanded_space.discretize(
+        tf.minimum(2000, 500 * tf.shape(expanded_space.lower)[-1])
+    )
 
-    bijector = tfp.bijectors.Sigmoid(low=space.lower, high=space.upper)
+    def vectorized_target_func(at):
+        return target_func(tf.reshape(at, at.shape[:-1].as_list() + [num_query_points, -1]))
```

Comment:
if this stuff is a separate function, it can be used with both `optimize_contrinuous` and `optimize_discrete`

Reply:
it also means we don't need to tinker with `optimize_continuous` and `optimize_discrete` in this PR

Reply:
and lets users define optimizers without needing to worry about batch dims. They can just wrap their optimizer in a convenient batchifier

Reply:
i'm not clear on this one: but it might also make the greedy-batch optimizer much easier to add

Reply:
Please elaborate. Do you have a clear idea how to do this?